lightning/src/offers/static_invoice.rs

```diff
@@ -158,6 +300,53 @@ impl StaticInvoice {
 }
 
 impl InvoiceContents {
+	#[cfg(feature = "std")]
+	fn is_offer_expired(&self) -> bool {
+		self.offer.is_expired()
+	}
+
+	#[cfg(not(feature = "std"))]
+	fn is_offer_expired_no_std(&self, duration_since_epoch: Duration) -> bool {
+		self.offer.is_expired_no_std(duration_since_epoch)
+	}
+
+	fn new(
+		offer: &Offer, payment_paths: Vec<(BlindedPayInfo, BlindedPath)>,
+		message_paths: Vec<BlindedPath>, created_at: Duration, signing_pubkey: PublicKey,
+	) -> Self {
+		Self {
+			offer: offer.contents.clone(),
+			payment_paths,
+			message_paths,
+			created_at,
+			relative_expiry: None,
+			fallbacks: None,
+			features: Bolt12InvoiceFeatures::empty(),
+			signing_pubkey,
+		}
+	}
+
+	fn as_invoice_fields_tlv_stream(&self) -> InvoiceTlvStreamRef {
```

Comment:
Could we include the offer TLV stream here, too, for consistency with the other code? (i.e., return a `PartialInvoiceTlvStreamRef`). I realize it won't be used later but it is nice for testing.